core/src/main/java/com/amplifyframework/core/model/temporal/Temporal.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model.temporal;
+
+import androidx.annotation.NonNull;
+import androidx.core.util.ObjectsCompat;
+
+import org.threeten.bp.DateTimeUtils;
+import org.threeten.bp.Instant;
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.LocalTime;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.OffsetTime;
+import org.threeten.bp.ZoneOffset;
+import org.threeten.bp.format.DateTimeFormatter;
+import org.threeten.bp.format.DateTimeFormatterBuilder;
+import org.threeten.bp.format.DateTimeParseException;
+import org.threeten.bp.temporal.ChronoField;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Namespace for all the Date/Time related types.
+ */
+public final class Temporal {
```

Comment:
I'm not crazy about having all of these crammed into one big file, though I understand there's not really another approach to achieve the naming strategy we agreed on.   

One option would be to call these `TemporalDate`, `TemporalDateTime`, ...  instead of `Temporal.Date` , `Temporal.DateTime`?   I understand we already made a decision on the naming though, so I'm fine with what's in this PR.

Reply:
I had the same conflict tbh. I didn't think about this when we decided on the naming, but then one I saw that it was less than 1k lines and the IDE helps you navigate I decided not to bring yet another naming discussion to the surface at this point...

Also, iOS has an even bigger PR with the `Temporal.*` name. The difference is that there we can do extensions like in Kotlin, so we can divide each temporal in its own file by adding an extension to the namespace class.